### PR TITLE
Implement fromBytes and toString for u256

### PR DESCRIPTION
**Pull request by bowenwang1996**
Implement `fromBytes` and `toString` for `u256`, as well as some other methods required to implement those two. Also add some tests for `u256`.

### CODE REVIEW

File: assembly/integer/u256.ts [hunk 1]
```diff
@@ -202,6 +227,79 @@ export class u256 {
...
+              return ah1 < bh1;
+          }
+      } else {
+          return ah2 < bh2;
+      }
+  }
+
+  @inline @operator('>')
+  static gt(a: u256, b: u256): bool {
+      return b < a;
```
Comment (MaxGraey) [inline assembly/integer/u256.ts]: nice!

File: assembly/integer/u256.ts [hunk 2]
```diff
@@ -202,6 +227,79 @@ export class u256 {
...
 
+  @inline @operator('>>')
+  static shr(value: u256, shift: i32): u256 {
+      shift &= 255;
+
+      // need for preventing redundant i32 -> u64 extends
+      var shift64 = shift as u64;
+      let lo1: u64, lo2: u64, hi1: u64, hi2: u64;
+
+      if (shift64 <= 64) {
```
Comment (MaxGraey) [inline assembly/integer/u256.ts]: I think better compare with 32-bit `shift` instead `shift64` (in other branches as well)

Comment (bowenwang1996) [reply assembly/integer/u256.ts]: Sure. Should I change the one in u128 implementation as well?

File: assembly/integer/u256.ts [hunk 3]
```diff
@@ -202,6 +227,79 @@ export class u256 {
...
+  }
+
+  @inline @operator('>>>')
+  static shr_u(value: u256, shift: i32): u256 {
+      return u256.shr(value, shift);
+  }
+
+  @inline @operator('==')
+  static eq(a: u256, b: u256): bool {
+      return a.hi2 == b.hi2 && a.hi1 == b.hi1 && a.lo2 == b.lo2 && a.lo1 == b.lo1;
```
Comment (MaxGraey) [inline assembly/integer/u256.ts]: plz reorder this comparison sequence from lower to highter

File: assembly/utils.ts [hunk 4]
```diff
@@ -130,19 +129,32 @@ export function utoa10(value: u128): string {
...
-    for (let digit_index = 0; digit_index < length; digit_index++) {
-      digits[digit_index] += digits[digit_index] >= 5 ? 3 : 0;
-    }
-    for (let j = length - 1; j != -1; j--) {
-      digits[j] <<= 1;
-      if (j < length - 1) digits[j + 1] |= digits[j] > 15 ? 1 : 0;
-      digits[j] &= 15;
-    }
-    digits[0] += <u8>left_bit;
+export function utoa10(value: u128): string {
```
Comment (MaxGraey) [inline assembly/utils.ts]: rename it to `u128toa10` for consistency

File: assembly/integer/u256.ts (hunk 2 above)
Comment (MaxGraey) [reply assembly/integer/u256.ts]: yes, indeed

**Comment by MaxGraey** [comment]
Ok, LGTM. Plz fix comments and reformat to using 2-spaced indent

File: assembly/integer/u256.ts (hunk 2 above)
Comment (bowenwang1996) [reply assembly/integer/u256.ts]: It seems that when I changed the line in `u128` to `shift32 = shift as u32`, many tests fail.

Comment (MaxGraey) [reply assembly/integer/u256.ts]: no, I mean keep `var shift64 = shift as u64;` line, but use `shift` param (which i32) during comparison, like:
```ts
if (shift <= 64) { // <- here use `shift` (i32 <= i32)
  hi2 = value.hi2 >> shift64; // <- here use shift64 (u64 >> u64)
  ...
  lo1 = (value.lo1 >> shift64) | (lo2 << (64 - shift64));
}
```

Comment (bowenwang1996) [reply assembly/integer/u256.ts]: I see

**Comment by bowenwang1996** [comment]
@MaxGraey I addressed the comments and reformatted using 2-space indent.

**Comment by MaxGraey** [comment]
Plz do this also for tests and specs =) This is last request and I'll merge it

File: tests/assembly/u256.spec.as.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+            0x99, 0xAA, 0xBB, 0xCC, 0xDD, 0xEE, 0xFF, 0x12,
+            0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88,
+            0x99, 0xAA, 0xBB, 0xCC, 0xDD, 0xEE, 0xFF, 0x12
+        ];
+
+        return u256.fromBytes(arr, true) == new u256(0x99aabbccddeeff12, 0x1122334455667788, 0x99aabbccddeeff12, 0x1122334455667788);
+    }
+
+    static shouldConvertToBytesLitteEndian(): bool {
+        // var a: u8[] = (new u128(0x8877665544332211, 0x12ffeeddccbbaa99)).toBytes();
```
Comment (MaxGraey) [inline tests/assembly/u256.spec.as.ts]: Also remove this comment

**Comment by bowenwang1996** [comment]
Sorry about that. I forgot :)

**Comment by MaxGraey** [comment]
Thanks!

**Pull request merged**